great_expectations/expectations/expectation.py

```diff
@@ -349,9 +355,17 @@ class Expectation(metaclass=MetaExpectation):
     expectation_type: ClassVar[str]
     examples: ClassVar[List[dict]] = []
 
-    def __init__(self, configuration: ExpectationConfiguration) -> None:
-        self._configuration = configuration
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+
+        # Everything below is purely to maintain current validation logic
+        expectation_type = camel_to_snake(self.__class__.__name__)
+        configuration = ExpectationConfiguration(
+            expectation_type=expectation_type,
+            kwargs=kwargs,
+        )
         self.validate_configuration(configuration)
+        self._configuration = configuration
```

Comment:
The idea is to eventually move the `self.validate_configuration` logic over to Pydantic validators.

I believe we'll need to keep `self._configuration` around for the time being as it is pretty deeply entrenched in various parts of the codebase; I will attempt to strip it out but only after we wrap up the primary effort.